File: web/src/types/ai-chat.ts
export interface AiChatMessage {
  id: number;
  conversation_id: number;
  role: "user" | "assistant" | "tool";
  content: string;
  tool_calls?: Record<string, unknown> | null;
  tool_call_id?: string | null;
  created_at: string;
}

export interface AiChatConversation {
  id: number;
  title: string;
  user_id: string;
  created_at: string;
  updated_at: string;
  message_count?: number;
  messages?: AiChatMessage[];
}

export interface AiChatConversationListResponse {
  items: AiChatConversation[];
  total: number;
}

export interface AiChatMessageResponse {
  message: AiChatMessage;
  reply: AiChatMessage;
}

File: api/app/schemas/ai_chat.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from .user import UserPublic


class AiChatMessageSummary(BaseModel):
    id: int
    conversation_id: int
    role: str
    content: str
    created_at: datetime


class AiChatConversationSummary(BaseModel):
    id: int
    title: str
    user_id: str
    created_at: datetime
    updated_at: datetime
    user: UserPublic | None = None
    message_count: int = 0


class AiChatConversationDetail(BaseModel):
    id: int
    title: str
    user_id: str
    created_at: datetime
    updated_at: datetime
    user: UserPublic | None = None
    messages: list[AiChatMessageSummary] = []


class AiChatConversationListResponse(BaseModel):
    items: list[AiChatConversationSummary]
    total: int


class AiChatConversationCreateRequest(BaseModel):
    title: str = Field(default="新对话", min_length=1, max_length=256)


class AiChatConversationUpdateRequest(BaseModel):
    title: str = Field(min_length=1, max_length=256)


class AiChatMessageSendRequest(BaseModel):
    content: str = Field(min_length=1, max_length=20000)


class AiChatMessageResponse(BaseModel):
    message: AiChatMessageSummary
    reply: AiChatMessageSummary
